test/sqllogictest/subquery.slt

```diff
@@ -321,15 +317,13 @@ EXPLAIN PLAN FOR SELECT *  FROM t1, t3 WHERE t1.a = t3.a AND EXISTS (SELECT * FR
 | | implementation = DeltaQuery
 | |   delta %4 %5.(#0)
 | |   delta %5 %4.(#0)
-| | demand = (#0)
 | Distinct group=(#0)
 | ArrangeBy (#0)
 
 %7 =
 | Join %3 %6 (= #2 #3)
 | | implementation = Differential %3 %6.(#0)
-| | demand = (#0, #2)
-| Project (#0, #0, #2)
+| Project (#0..#2)
```

Comment:
Flagging as interesting. Join stuff puts in a projection `Project (#0, #0, #2)` in an attempt to remove demand for equated columns. We should check that pushing that projection down doesn't cause harm elsewhere.